phalcon/forms/form.zep

```diff
@@ -369,51 +362,30 @@ class Form extends Injectable implements \Countable, \Iterator
 		/**
 		 * Return the validation status
 		 */
-		return notFailed;
+		return validationStatus;
 	}
 
 	/**
 	 * Returns the messages generated in the validation
 	 */
 	public function getMessages(boolean byItemName = false) -> <Group>
 	{
-		var messages, group, elementMessages;
+		var messages;
 
 		let messages = this->_messages;
-		if byItemName {
-			if typeof messages != "array" {
-				return new Group();
-			}
-			return messages;
-		}
-
-		let group = new Group();
-
-		if typeof messages == "array" {
-			for elementMessages in messages {
-				group->appendMessages(elementMessages);
-			}
+		if typeof messages == "object" && messages instanceof Group {
+            return messages;
 		}
 
-		return group;
+		return new Group();
 	}
 
 	/**
 	 * Returns the messages generated for a specific element
 	 */
 	public function getMessagesFor(string! name) -> <Group>
 	{
-		var messages, elementMessages, group;
-
-		let messages = this->_messages;
-		if fetch elementMessages, messages[name] {
-			return elementMessages;
-		}
-
-		let group = new Group(),
-			this->_messages[name] = group;
-
-		return group;
+        return this->get(name)->getMessages();
```

Comment:
Well, in general looks good to me. But it bothers me that previously this method always returned the `Group` but now it can throw Exception. I think this behaviour should be the same now.